### PR TITLE
Feature/grch38 2

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Grch38 features but is still a work in progress

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: templates/mip_download_rd_dna_config_-1.0-.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 bash_set_errexit: 1
 bash_set_pipefail: 1
 load_env:
-   MIP_rd_dna:
+   D_mip7.0_190503_hs:
```
Comment (henrikstranneheim) [inline templates/mip_download_rd_dna_config_-1.0-.yaml]:
```suggestion
   MIP_rd_dna:
```

File: templates/mip_download_rd_dna_config_-1.0-.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
+#   MIP_rd_dna:
```
Comment (henrikstranneheim) [inline templates/mip_download_rd_dna_config_-1.0-.yaml]:
```suggestion
```

File: lib/MIP/Language/Perl.pm [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    require Exporter;
+    use base qw{ Exporter };
+
+    # Set the version for version checking
+    our $VERSION = 1.00;
+
+    our @EXPORT_OK = qw{ perl_base perl_nae_oneliners };
+}
+
+sub perl_base {
```
Comment (jemten) [inline lib/MIP/Language/Perl.pm]: Nice! No we can start adding that to our hard coded perl oneliners :)

File: lib/MIP/Language/Perl.pm [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    ## Flatten argument(s)
+
+    ## Default(s)
+    my $autosplit;
+    my $command_line;
+    my $n;
+
+    my $tmpl = {
+        autosplit => {
+            allow       => qr{ \A\d+\z }sxm,
```
Comment (jemten) [inline lib/MIP/Language/Perl.pm]: Will this ever have a value of something else than 0 or 1?

File: lib/MIP/Language/Perl.pm [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    }
+    if ($command_line) {
+
+        push @commands, q{-e};
+    }
+
+    return @commands;
+}
+
+sub perl_nae_oneliners {
```
Comment (jemten) [inline lib/MIP/Language/Perl.pm]: Is the idea to add all new oneliners here? In that case will I add a new variable for each one liner?

File: t/perl_nae_oneliners.t [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+use FindBin qw{ $Bin };
+use open qw{ :encoding(UTF-8) :std };
+use Params::Check qw{ allow check last_error };
+use Test::More;
+use utf8;
+use warnings qw{ FATAL utf8 };
+
+## CPANM
+use autodie qw{ :all };
+use Modern::Perl qw{ 2014 };
```
Comment (jemten) [inline t/perl_nae_oneliners.t]:
```suggestion
use Modern::Perl qw{ 2018 };
```

File: t/perl_base.t [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+use FindBin qw{ $Bin };
+use open qw{ :encoding(UTF-8) :std };
+use Params::Check qw{ allow check last_error };
+use Test::More;
+use utf8;
+use warnings qw{ FATAL utf8 };
+
+## CPANM
+use autodie qw{ :all };
+use Modern::Perl qw{ 2014 };
```
Comment (jemten) [inline t/perl_base.t]:
```suggestion
use Modern::Perl qw{ 2018 };
```

File: t/download_cadd_offline_annotations.t [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+use FindBin qw{ $Bin };
+use open qw{ :encoding(UTF-8) :std };
+use Params::Check qw{ allow check last_error };
+use Test::More;
+use utf8;
+use warnings qw{ FATAL utf8 };
+
+## CPANM
+use autodie qw { :all };
+use Modern::Perl qw{ 2014 };
```
Comment (jemten) [inline t/download_cadd_offline_annotations.t]:
```suggestion
use Modern::Perl qw{ 2018 };
```

File: t/analysis_cadd_gb_38.t [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+use FindBin qw{ $Bin };
+use open qw{ :encoding(UTF-8) :std };
+use Params::Check qw{ allow check last_error };
+use Test::More;
+use utf8;
+use warnings qw{ FATAL utf8 };
+
+## CPANM
+use autodie qw { :all };
+use Modern::Perl qw{ 2014 };
```
Comment (jemten) [inline t/analysis_cadd_gb_38.t]:
```suggestion
use Modern::Perl qw{ 2018 };
```

File: t/set_recipe_cadd.t [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+use FindBin qw{ $Bin };
+use open qw{ :encoding(UTF-8) :std };
+use Params::Check qw{ allow check last_error };
+use Test::More;
+use utf8;
+use warnings qw{ FATAL utf8 };
+
+## CPANM
+use autodie qw { :all };
+use Modern::Perl qw{ 2014 };
```
Comment (jemten) [inline t/set_recipe_cadd.t]:
```suggestion
use Modern::Perl qw{ 2018 };
```

File: lib/MIP/Language/Perl.pm (hunk 4 above)
Comment (henrikstranneheim) [reply lib/MIP/Language/Perl.pm]: true that I can change it to boolean instead of a regexp

File: lib/MIP/Language/Perl.pm (hunk 5 above)
Comment (henrikstranneheim) [reply lib/MIP/Language/Perl.pm]: yes, currently. We could supply a list of keys and then loop around them in hash map of predefined keys to oneliners. It need to work with the test of not collapsing for the same key so they need to be unique. However, it is a bit uggly to supply a list when the logical thing is a single key. But perhaps better than an ever growing parameter list. I will try it

File: lib/MIP/Language/Perl.pm (hunk 3 above)
Comment (henrikstranneheim) [reply lib/MIP/Language/Perl.pm]: 👍

**Pull request merged**